lib/spack/spack/binary_distribution.py

```diff
@@ -743,8 +753,18 @@ def get_buildfile_manifest(spec):
                 continue
 
         elif relocate.needs_text_relocation(m_type, m_subtype):
-            data["text_to_relocate"].append(rel_path)
-            continue
+            added = False
+            with io.open(abs_path, "rb") as f:
+                contents = f.read()
+
+            for prefix in compiled_prefixes:
+                if prefix.search(contents):
+                    data["text_to_relocate"].append(rel_path)
+                    added = True
+                    break
+
+            if added:
+                continue
```

Comment:
Can you please make it use something like to get rid of this awkward flow and five levels of indentation?

```python
def file_matches_any_binary_regex(path, regexes):
    with open(path, "rb") as f:
        contents = f.read()
    for regex in regexes:
        if regex.search(contents):
            return True
    return False
```

Reply:
Or just use `any` w/o a function, but anything that does not have `added = True`...